CONTRIBUTING.md

```diff
@@ -0,0 +1,134 @@
+Executing all these tests requires to install and configure several software packages:
+- [Docker](https://docs.docker.com/engine/install/), which is required to test the `docker` and `ssh` connectors and the CWL `DockerRequirement` implementation with Docker containers;
+- [Singularity](https://docs.sylabs.io/guides/3.0/user-guide/installation.html), which is required to test the `singularity` connector and the CWL `DockerRequirement` conversion to Singularity containers;
+- A Kubernetes distribution, e.g., [minikube](https://minikube.sigs.k8s.io/docs/start/), which is used to test the `kubernetes` connector and the CWL `DockerRequirement` conversion to Kubernetes pods.
+
+For development purposes, it is possible to execute local tests only on a specific subset of connectors using the `--deploys` option as follows
+```bash 
+pytest --deploys local,docker,singularity tests/test_remotepath.py
+```
+However, all connectors are always tested in the CI pipeline.
+
+
+
+### Code style
+StreamFlow supports Python 3.8 to 3.12, so use compatible code features.
+
+The StreamFlow code style complies with [PEP 8](https://peps.python.org/pep-0008/).
+Some automatic style checkers can be installed with
+```bash
+pip install -r lint-requirements.txt
+```
+
+You can verify that your PR respects the code style with the following command
+```bash
+make format-check flake8 codespell-check
+```
+
+Otherwise, you can apply automatically the suggested changes. Use them carefully because they will write the file back. 
+```bash
+make format codespell
+```
+
+
+
+
+
+## Contribute to the documentation
+
+### Generate the documentation
+The documentation is written in the reStructuredText markup language, and the [Sphinx](https://www.sphinx-doc.org/en/master/) framework is used to generate the HTML pages published on the [StreamFlow website](https://streamflow.di.unito.it/documentation/latest/). Currently, there is no CD for documentation: maintainers manually transfer files on the StreamFlow website at every release.
+
+To check your local version of the documentation, first, install all the required packages using the following command
+```bash
+cd docs
+pip install -r requirements.txt
+```
+
+Then, generate the documentation in HTML format with the following command
+```bash
+make html
+```
+
+If you want, you can also generate documentation in a different format from the ones supported by Sphinx. However, note that StreamFlow officially maintains only the HTML format. For more information on the available documentation formats, use the following command
+```bash
+make help
+```
+
+### Documentation changes
+For any changes to the documentation, it is necessary to update the dedicated test on the CI pipeline.
+In particular, the checksum of the documentation in the HTML format is tested to avoid misleading changes when new Sphinx versions are released.
```

Comment:
```suggestion
The checksum of the documentation in HTML format is tested in the CI pipeline to avoid unwanted changes when Shpinx or other dependencies are updated.
```